src/components/logger/Logger.js

```diff
@@ -0,0 +1,200 @@
+// @flow
+import * as React from 'react';
+import noop from 'lodash/noop';
+import { METRIC_TYPE_PREVIEW_METRIC, METRIC_TYPE_ELEMENTS_LOAD_METRIC } from '../../constants';
+
+type ElementsMetric = {
```

Comment:
It looks like `ElementsMetricCallback` is defined in the top level flowtypes file. Should these type declarations live together?

Reply:
It could, but I put it here to avoid bloating the already large `box-ui-elements.js` since its used only in this file (for now), similar to how we use `constants.js`